src/main/java/org/candlepin/model/Consumer.java

```diff
@@ -1024,4 +988,32 @@ public Consumer setServiceType(String serviceType) {
         this.serviceType = serviceType;
         return this;
     }
+
+    public Map<String, String> getEnvironmentIds() {
+        return this.environmentIds;
```

Comment:
The internal collection should not be returned directly. Hibernate uses proxy collections that can cause unintended modifications to an object if the collection is reused or modified externally.

We need to encapsulate this properly, both protecting the collection and hiding the implementation details of how we're storing this in the DB.

Reply:
Now wrapped in MapView.